### PR TITLE
[New Rule] Suspicious DLL Loaded for Persistence or Privilege Escalation

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/818

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/privilege_escalation_persistence_phantom_dll.toml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    "http://waleedassar.blogspot.com/2013/01/wow64logdll.html",
+]
+risk_score = 73
+rule_id = "bfeaf89b-a2a7-48a3-817f-e41829dc61ee"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Persistence", "Privilege Escalation"]
+type = "eql"
+
+query = '''
+library where dll.name:
```
Comment (bm11100) [inline rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: just a spacing nitpick - 

```
library where dll.name:
  (
  "wlbsctrl.dll",
  "wbemcomn.dll",
  "WptsExtensions.dll",
  "Tsmsisrv.dll",
  "TSVIPSrv.dll",
  "Msfte.dll",
  "wow64log.dll",
  "WindowsCoreDeviceInfo.dll",
  "Ualapi.dll",
  "wlanhlp.dll",
  "phoneinfo.dll",
  "EdgeGdi.dll",
  "cdpsgshims.dll",
  "windowsperformancerecordercontrol.dll",
  "diagtrack_win.dll"
  ) and not
(dll.code_signature.subject_name:"Microsoft Windows" and dll.code_signature.status:"trusted")
```

Comment (bm11100) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: I'm not getting CI errors but in dev tools it looks like there's a verification exception on both the dll code signature fields, i.e `Unknown column [dll.code_signature.subject_name]`

If we try Ext the [keyword] type seems to be an unsupported nested type in hierarchy

Comment (bm11100) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: image

image

image

Comment (bm11100) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: Is `dll.code_signature.*` supported in 7.11 for library events? @rw-access

Comment (bm11100) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: I'm testing on 7.11 and looks like both `dll.code_signature.subject_name` and `dll.code_signature.status` are giving me a verification exception shown in the first screenshot above

Comment (rw-access) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: ah. can you check the mapping to see if that field is in it?

Comment (rw-access) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: looks like `dll.code_signature` isn't in the mapping

Comment (rw-access) [reply rules/windows/privilege_escalation_persistence_phantom_dll.toml:33]: https://github.com/elastic/package-storage/blob/production/packages/endpoint/0.17.0/data_stream/library/fields/fields.yml#L153

File: rules/windows/privilege_escalation_persistence_phantom_dll.toml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+[metadata]
+creation_date = "2020/01/07"
+maturity = "production"
+updated_date = "2020/01/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the loading of a non Microsoft signed DLL that is missing on a default Windows install (phantom DLL) or one that can be loaded from a different location by a native Windows process. This may be abused to persist or elevate privileges via privileged file write vulnerabilities.
```
Comment (bm11100) [inline rules/windows/privilege_escalation_persistence_phantom_dll.toml:9]: re-lint

File: rules/windows/privilege_escalation_persistence_phantom_dll.toml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  "WindowsCoreDeviceInfo.dll",
+  "Ualapi.dll",
+  "wlanhlp.dll",
+  "phoneinfo.dll",
+  "EdgeGdi.dll",
+  "cdpsgshims.dll",
+  "windowsperformancerecordercontrol.dll",
+  "diagtrack_win.dll"
+  ) and not
+(dll.code_signature.subject_name:"Microsoft Windows" and dll.code_signature.status:"trusted")
```
Comment (bm11100) [inline rules/windows/privilege_escalation_persistence_phantom_dll.toml:49]:
```suggestion
  ) and 
not (dll.code_signature.subject_name:"Microsoft Windows" and dll.code_signature.status:"trusted")
```

**Review by bm11100 — APPROVED** [review]
As long as https://github.com/elastic/endpoint-package/pull/131 takes care of the `dll.code_signature` mapping issue, LGTM

File: rules/windows/privilege_escalation_persistence_phantom_dll.toml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  "WindowsCoreDeviceInfo.dll",
+  "Ualapi.dll",
+  "wlanhlp.dll",
+  "phoneinfo.dll",
+  "EdgeGdi.dll",
+  "cdpsgshims.dll",
+  "windowsperformancerecordercontrol.dll",
+  "diagtrack_win.dll"
+  ) and 
+not (dll.code_signature.subject_name:"Microsoft Windows" and dll.code_signature.status:"trusted")
```
Comment (bm11100) [inline rules/windows/privilege_escalation_persistence_phantom_dll.toml:51]:
```suggestion
library where dll.name :
  (
  "wlbsctrl.dll",
  "wbemcomn.dll",
  "WptsExtensions.dll",
  "Tsmsisrv.dll",
  "TSVIPSrv.dll",
  "Msfte.dll",
  "wow64log.dll",
  "WindowsCoreDeviceInfo.dll",
  "Ualapi.dll",
  "wlanhlp.dll",
  "phoneinfo.dll",
  "EdgeGdi.dll",
  "cdpsgshims.dll",
  "windowsperformancerecordercontrol.dll",
  "diagtrack_win.dll"
  ) and 
not (dll.code_signature.subject_name : "Microsoft Windows" and dll.code_signature.status : "trusted")
```

File: rules/windows/privilege_escalation_persistence_phantom_dll.toml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the loading of a non Microsoft signed DLL that is missing on a default Windows install (phantom DLL) or one
+that can be loaded from a different location by a native Windows process. This may be abused to persist or elevate
+privileges via privileged file write vulnerabilities.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_persistence_phantom_dll.toml:14]:
```suggestion
index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
```

**Review by brokensound77 — APPROVED** [review]
Looks like the package mapping merged - just the windows index and LGTM 👍

**Pull request merged**